Tests/MapboxNavigationTests/VanishingRouteLineTests.swift

```diff
@@ -0,0 +1,249 @@
+import XCTest
+import MapboxDirections
+import TestHelper
+import Turf
+import MapboxMaps
+@testable import MapboxNavigation
+@testable import MapboxCoreNavigation
+
+class VanishingRouteLineTests: TestCase {
+    var navigationMapView: NavigationMapView!
+    
+    override func setUp() {
+        super.setUp()
+        navigationMapView = NavigationMapView(frame: CGRect(origin: .zero, size: .iPhone6Plus))
+    }
+    
+    override func tearDown() {
+        navigationMapView = nil
+        super.tearDown()
+    }
+    
+    func getRoute() -> Route {
+        let routeData = Fixture.JSONFromFileNamed(name: "short_route")
+        let routeOptions = NavigationRouteOptions(coordinates: [
+            CLLocationCoordinate2DMake(-122.5237429, 37.975393),
+            CLLocationCoordinate2DMake(-122.5231413, 37.9750695)
+        ])
+        let decoder = JSONDecoder()
+        decoder.userInfo[.options] = routeOptions
+        var testRoute: Route?
+        XCTAssertNoThrow(testRoute = try decoder.decode(Route.self, from: routeData))
+        guard let route = testRoute else {
+            preconditionFailure("Route is invalid.")
+        }
+        
+        return route
+    }
+    
+    func getMultilegRoute() -> Route {
+        let routeData = Fixture.JSONFromFileNamed(name: "multileg_route")
+        let routeOptions = NavigationRouteOptions(coordinates: [
+            CLLocationCoordinate2DMake(-77.1576396, 38.7830304),
+            CLLocationCoordinate2DMake(-77.1670888, 38.7756155),
+            CLLocationCoordinate2DMake(-77.1534183, 38.7708948),
+        ])
+        let decoder = JSONDecoder()
+        decoder.userInfo[.options] = routeOptions
+        var testRoute: Route?
+        XCTAssertNoThrow(testRoute = try decoder.decode(Route.self, from: routeData))
+        guard let route = testRoute else {
+            preconditionFailure("Route is invalid.")
+        }
+        
+        return route
+    }
+    
+    func getRouteProgress() -> RouteProgress {
+        let route = getRoute()
+        let routeProgress = RouteProgress(route: route, options: routeOptions, legIndex: 0, spokenInstructionIndex: 0)
+        routeProgress.currentLegProgress = RouteLegProgress(leg: route.legs[0], stepIndex: 2, spokenInstructionIndex: 0)
+        routeProgress.currentLegProgress.currentStepProgress = RouteStepProgress(step: route.legs[0].steps[2], spokenInstructionIndex: 0)
+        return routeProgress
+    }
+    
+    func linaGradientToString(lineGradient: Value<StyleColor>?) -> String {
+        return lineGradient.debugDescription.components(separatedBy: "(").last!.components(separatedBy: ")").first!
```

Comment:
Nit: even tough using force unwrapping in tests can be sometimes useful to see errors early, while running in Xcode it leads to undesired stops while running all tests. I'd recommend to remove force unwrapping and add assert, in case of string is invalid.

Reply:
Thanks for the suggestion, I'll add the assertion for helping methods in tests.